config/config.go

```diff
@@ -1157,6 +1122,91 @@ func migrateConfigTCF2PurposeEnabledFlags(v *viper.Viper) {
 	}
 }
 
+func migrateConfigDatabaseConnection(v *viper.Viper) {
+
+	migrations := []struct {
+		old    string
+		new    string
+		fields []string
+	}{
+		{
+			old: "stored_requests.postgres",
+			new: "stored_requests.database",
+			fields: []string{
+				"connection.dbname",
+				"connection.host",
+				"connection.port",
+				"connection.user",
+				"connection.password",
+				"fetcher.query",
+				"fetcher.amp_query",
+				"initialize_caches.timeout_ms",
+				"initialize_caches.query",
+				"initialize_caches.amp_query",
+				"poll_for_updates.refresh_rate_seconds",
+				"poll_for_updates.timeout_ms",
+				"poll_for_updates.query",
+				"poll_for_updates.amp_query",
+			},
+		},
+		{
+			old: "stored_video_req.postgres",
+			new: "stored_video_req.database",
+			fields: []string{
+				"connection.dbname",
+				"connection.host",
+				"connection.port",
+				"connection.user",
+				"connection.password",
+				"fetcher.query",
+				"initialize_caches.timeout_ms",
+				"initialize_caches.query",
+				"poll_for_updates.refresh_rate_seconds",
+				"poll_for_updates.timeout_ms",
+				"poll_for_updates.query",
+			},
+		},
+		{
+			old: "stored_responses.postgres",
+			new: "stored_responses.database",
+			fields: []string{
+				"connection.dbname",
+				"connection.host",
+				"connection.port",
+				"connection.user",
+				"connection.password",
+				"fetcher.query",
+				"initialize_caches.timeout_ms",
+				"initialize_caches.query",
+				"poll_for_updates.refresh_rate_seconds",
+				"poll_for_updates.timeout_ms",
+				"poll_for_updates.query",
+			},
+		},
+	}
+
+	for _, migration := range migrations {
+		driverField := migration.new + ".connection.driver"
+		if !v.IsSet(driverField) {
+			glog.Warning(fmt.Sprintf("%s is not set, using default (postgres) ", driverField))
+			v.Set(driverField, "postgres")
+		}
```

Comment:
Perhaps it makes sense to only ever show this warning if we believe the host intends to use a database for stored requests to reduce noise and confusion on startup. We could lean on the old postgres `dbname` config field to make that determination:
```
		if !v.IsSet(driverField) && v.IsSet("postgres.connection.dbname") {
			glog.Warning(fmt.Sprintf("%s is not set, using default (postgres) ", driverField))
			v.Set(driverField, "postgres")
		}
```

Reply:
Fixed